File: android/src/main/java/com/spoonacular/client/model/AddMealPlanTemplate200ResponseItemsInnerValue.java
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 800,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

package com.spoonacular.client.model;

import java.math.BigDecimal;
import io.swagger.annotations.*;
import com.google.gson.annotations.SerializedName;

@ApiModel(description = "")
public class AddMealPlanTemplate200ResponseItemsInnerValue {
  
  @SerializedName("id")
  private Integer id = null;
  @SerializedName("servings")
  private BigDecimal servings = null;
  @SerializedName("title")
  private String title = null;
  @SerializedName("imageType")
  private String imageType = null;

  /**
   **/
  @ApiModelProperty(value = "")
  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public BigDecimal getServings() {
    return servings;
  }
  public void setServings(BigDecimal servings) {
    this.servings = servings;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getTitle() {
    return title;
  }
  public void setTitle(String title) {
    this.title = title;
  }

  /**
   **/
  @ApiModelProperty(value = "")
  public String getImageType() {
    return imageType;
  }
  public void setImageType(String imageType) {
    this.imageType = imageType;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AddMealPlanTemplate200ResponseItemsInnerValue addMealPlanTemplate200ResponseItemsInnerValue = (AddMealPlanTemplate200ResponseItemsInnerValue) o;
    return (this.id == null ? addMealPlanTemplate200ResponseItemsInnerValue.id == null : this.id.equals(addMealPlanTemplate200ResponseItemsInnerValue.id)) &&
        (this.servings == null ? addMealPlanTemplate200ResponseItemsInnerValue.servings == null : this.servings.equals(addMealPlanTemplate200ResponseItemsInnerValue.servings)) &&
        (this.title == null ? addMealPlanTemplate200ResponseItemsInnerValue.title == null : this.title.equals(addMealPlanTemplate200ResponseItemsInnerValue.title)) &&
        (this.imageType == null ? addMealPlanTemplate200ResponseItemsInnerValue.imageType == null : this.imageType.equals(addMealPlanTemplate200ResponseItemsInnerValue.imageType));
  }

  @Override
  public int hashCode() {
    int result = 17;
    result = 31 * result + (this.id == null ? 0: this.id.hashCode());
    result = 31 * result + (this.servings == null ? 0: this.servings.hashCode());
    result = 31 * result + (this.title == null ? 0: this.title.hashCode());
    result = 31 * result + (this.imageType == null ? 0: this.imageType.hashCode());
    return result;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class AddMealPlanTemplate200ResponseItemsInnerValue {\n");
    
    sb.append("  id: ").append(id).append("\n");
    sb.append("  servings: ").append(servings).append("\n");
    sb.append("  title: ").append(title).append("\n");
    sb.append("  imageType: ").append(imageType).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

File: javascript/src/com.spoonacular.client/com.spoonacular.client.model/AddMealPlanTemplate200ResponseItemsInnerValue.js
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 800,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';

/**
 * The AddMealPlanTemplate200ResponseItemsInnerValue model module.
 * @module com.spoonacular.client/com.spoonacular.client.model/AddMealPlanTemplate200ResponseItemsInnerValue
 * @version 1.1
 */
class AddMealPlanTemplate200ResponseItemsInnerValue {
    /**
     * Constructs a new <code>AddMealPlanTemplate200ResponseItemsInnerValue</code>.
     * @alias module:com.spoonacular.client/com.spoonacular.client.model/AddMealPlanTemplate200ResponseItemsInnerValue
     */
    constructor() { 
        
        AddMealPlanTemplate200ResponseItemsInnerValue.initialize(this);
    }

    /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
    static initialize(obj) { 
    }

    /**
     * Constructs a <code>AddMealPlanTemplate200ResponseItemsInnerValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:com.spoonacular.client/com.spoonacular.client.model/AddMealPlanTemplate200ResponseItemsInnerValue} obj Optional instance to populate.
     * @return {module:com.spoonacular.client/com.spoonacular.client.model/AddMealPlanTemplate200ResponseItemsInnerValue} The populated <code>AddMealPlanTemplate200ResponseItemsInnerValue</code> instance.
     */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new AddMealPlanTemplate200ResponseItemsInnerValue();

            if (data.hasOwnProperty('id')) {
                obj['id'] = ApiClient.convertToType(data['id'], 'Number');
            }
            if (data.hasOwnProperty('servings')) {
                obj['servings'] = ApiClient.convertToType(data['servings'], 'Number');
            }
            if (data.hasOwnProperty('title')) {
                obj['title'] = ApiClient.convertToType(data['title'], 'String');
            }
            if (data.hasOwnProperty('imageType')) {
                obj['imageType'] = ApiClient.convertToType(data['imageType'], 'String');
            }
        }
        return obj;
    }


}

/**
 * @member {Number} id
 */
AddMealPlanTemplate200ResponseItemsInnerValue.prototype['id'] = undefined;

/**
 * @member {Number} servings
 */
AddMealPlanTemplate200ResponseItemsInnerValue.prototype['servings'] = undefined;

/**
 * @member {String} title
 */
AddMealPlanTemplate200ResponseItemsInnerValue.prototype['title'] = undefined;

/**
 * @member {String} imageType
 */
AddMealPlanTemplate200ResponseItemsInnerValue.prototype['imageType'] = undefined;






export default AddMealPlanTemplate200ResponseItemsInnerValue;

